app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -3,6 +3,7 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   def google_oauth2
     if user.present?
+      user.accept_invitation! if invitation_token
```

Comment:
Hmm, what happens if user.accept_invitation doesn't execute successfully? I think we should still sign in the user via Google OAuth but I'm wondering if it's worth displaying some sort of error message telling them the invitation didn't go through 🤔

Reply:
`accept_invitation!` just sets the `invitation_accepted_at` and clears the token so if the user is valid upto that point, it will not error out. The method does not raise an exception(although the `!` at the end of it is misleading), so the the user sign in will go through.

Reply:
There however is one catch here. Since the user can use any google account to login, the email for the google account used can be different from the email used in the invitation. I've not put in any restrictions so the user is able to sign-up even if that's the case.

Reply:
I think that's fine, it's possible that the invitation was originally sent to a non-Google account and the person wants to use their Google account instead. Thanks for explaining. 

Yeah the `!` is misleading :/ What happens if you exclude it?

Reply:
That's just how the method is named in `devise_invitable`. Excluding the `!` calls the method which sets the `invitation_accepted_at` but doesn't save the model.

Reply:
Ah okay, let's keep what we have here then! Thanks for clarifying!